src/core/TESTING.md

```diff
@@ -59,11 +59,294 @@ test('my test', async () => {
 });
 ```
 
-### Strategies for specific Core APIs
+## Strategies for specific Core APIs
+
+### HTTP Routes
+The HTTP API interface is another public contract of Kibana, although not every Kibana endpoint is for external use. When evaluating the required level of test coverage for an HTTP resource, make your judgment based on whether an endpoint is considered to be public or private. Public API is expected to have a higher level of test coverage.
+Public API tests should cover the **observable behavior** of the system, therefore they should be close to the real user interactions as much as possible, ideally by using HTTP requests to communicate with the Kibana server as a real user would do.
+
+#### Integration tests
```

Comment:
If you're starting to write tests you should start with unit tests first, so maybe we should reverse the order of the sections to do unit tests first?

Reply:
done